File: 3423.maximum-difference-between-adjacent-elements-in-a-circular-array/Solution.c
// @leet start
int abs(int a) { return (a > -1) ? a : (a * -1); }
int maxAB(int a, int b) { return (a > b) ? a : b; }
int maxAdjacentDistance(int *nums, int numsSize) {
  int max = abs(nums[0] - nums[numsSize - 1]);
  for (int i = 0, len = numsSize - 1; i < len; i++) {
    max = maxAB(max, abs(nums[i] - nums[i + 1]));
  }
  return max;
}
// @leet end

File: 3423.maximum-difference-between-adjacent-elements-in-a-circular-array/Solution.java
// @leet start
class Solution {
  public int maxAdjacentDistance(int[] nums) {
    int max = Math.abs(nums[0] - nums[nums.length - 1]);
    for (int i = 0, len = nums.length - 1; i < len; i++) {
      max = Math.max(max, Math.abs(nums[i] - nums[i + 1]));
    }
    return max;
  }
}
// @leet end
